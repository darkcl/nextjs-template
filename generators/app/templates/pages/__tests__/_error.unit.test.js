import React from 'react';
import toJson from 'enzyme-to-json';
import { render } from 'enzyme';

import Error from '../_error';

describe('<Error />', () => {
  it('getInitialProps should return object with status code', () => {
    let props = Error.getInitialProps({ res: { statusCode: 404 } });
    expect(props).toEqual({ statusCode: 404 });

    props = Error.getInitialProps({ err: { statusCode: 404 } });
    expect(props).toEqual({ statusCode: 404 });

    props = Error.getInitialProps({
      res: { statusCode: 404 },
      err: { statusCode: 500 },
    });

    expect(props).toEqual({ statusCode: 404 });
  });

  it('should render correctly without status code', () => {
    const document = render(<Error />);
    expect(toJson(document)).toMatchSnapshot();
  });

  it('should render correctly with status code', () => {
    const document = render(<Error statusCode={404} />);
    expect(toJson(document)).toMatchSnapshot();
  });
});
