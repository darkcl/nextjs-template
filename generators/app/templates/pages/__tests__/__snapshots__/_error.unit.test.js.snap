// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Error /> should render correctly with status code 1`] = `
Array [
  <div
    class="header"
  >
    <div
      class="container"
    >
      <a
        href="/"
      >
        <img
          src="undefined/static/47cd76e43f74bbc2e1baaf194d07e1fa/images/favicon.png"
        />
      </a>
      <span>
        header.title
      </span>
    </div>
  </div>,
  <div
    class="container"
  >
    <p>
      An error 404 occurred on server
    </p>
  </div>,
  <div
    class="footer"
  >
    <div
      class="container"
    >
      footer.text
    </div>
  </div>,
]
`;

exports[`<Error /> should render correctly without status code 1`] = `
Array [
  <div
    class="header"
  >
    <div
      class="container"
    >
      <a
        href="/"
      >
        <img
          src="undefined/static/47cd76e43f74bbc2e1baaf194d07e1fa/images/favicon.png"
        />
      </a>
      <span>
        header.title
      </span>
    </div>
  </div>,
  <div
    class="container"
  >
    <p>
      An error occurred on client
    </p>
  </div>,
  <div
    class="footer"
  >
    <div
      class="container"
    >
      footer.text
    </div>
  </div>,
]
`;
